contracts/FundMe.sol

```diff
@@ -45,24 +45,19 @@ contract FundMe {
         funders.push(msg.sender); //1TODO: one address can only fund once. Throw an error when trying to fund for the second time
     }
 
-    // 1ANS: This doesnt work (after I add this modifier, my tests fails)
-    // modifier fundOnce() {
-    //     for (
-    //         uint256 funderIndex = 0;
-    //         funderIndex < funders.length;
-    //         funderIndex++
-    //     ) {
-    //         if (funders[funderIndex] == msg.sender)
-    //             revert FundMe__AlreadyFunded();
-    //         _;
-    //     }
-    // }
+    // 1ANS: This should work
+    modifier fundOnce() {
+        if (addressToAmountFunded[msg.sender] != 0)
+            revert FundMe__AlreadyFunded();
+        _;
+    }
```

Comment:
Why is it a modifier? What is the alternative option to writing this? When do you use a modifier over the alternative option?